extensions/amp-geo/0.1/amp-geo.js

```diff
@@ -169,32 +171,35 @@ export class AmpGeo extends AMP.BaseElement {
    * @param {Document} doc
    */
   findCountry_(doc) {
-    // First see if we've been pre-rendered with a country, if so set it
+    // Flag to see if we've been pre-rendered with a country
     const preRenderMatch = doc.body.className.match(PRE_RENDER_REGEX);
+    // Trim the spaces off the patched country
+    const trimmedCountry = COUNTRY.trim();
 
-    if (preRenderMatch &&
-        !Services.urlForDoc(this.element).isProxyOrigin(doc.location)) {
-      this.mode_ = mode.GEO_PRERENDER;
-      this.country_ = preRenderMatch[1];
-    } else {
-      this.mode_ = mode.GEO_HOT_PATCH;
-      this.country_ = COUNTRY.trim();
-      // If we got a country code it will be 2 characters
-      // If the lengths is 0 the country is unknown
-      // If the length is > 2 we didn't get patched
-      // (probably local dev) so we treat it as unknown.
-      if (this.country_.length !== 2) {
-        this.country_ = 'unknown';
-      }
-    }
+    // default country is 'unknown' which is also the zero length case
 
-    // Are we in debug override?
-    // match to \w characters only to prevent xss vector
     if (getMode(this.win).geoOverride &&
       (isCanary(this.win) || getMode(this.win).localDev) &&
       /^\w+$/.test(getMode(this.win).geoOverride)) {
+      // debug override case, only works in canary or localdev
+      // match to \w characters only to prevent xss vector
       this.mode_ = mode.GEO_OVERRIDE;
       this.country_ = getMode(this.win).geoOverride.toLowerCase();
+    } else if (preRenderMatch &&
+        !Services.urlForDoc(this.element).isProxyOrigin(doc.location)) {
```

Comment:
Wait, shouldn't this be `preRenderMatch && isProxyOrigin` (no `!`)?

Reply:
Hmm, you're right - this was copied over from existing code. The bug has been there since launch but latent because we're not actually pre-rendering the geo classes. I'll fix it.

Reply:
Ah now I see it's the comment that's wrong.  The idea is that a publisher can pre render in a country code using their own geolocation but, if it's a cache we'll override it.  I'll update the comment to explain.